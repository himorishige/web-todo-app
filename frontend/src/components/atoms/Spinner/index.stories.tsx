/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Spinner from '.';

export default {
  title: 'WebToDoApp/Atoms/Spinner',
  component: Spinner,
} as ComponentMeta<typeof Spinner>;

const Template: ComponentStory<typeof Spinner> = (args) => <Spinner {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Spinner';
