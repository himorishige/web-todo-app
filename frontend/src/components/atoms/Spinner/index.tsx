/** @jsxImportSource @emotion/react */
import Loader from 'react-loader-spinner';

const Skelton: React.VFC = () => {
  return (
    <div>
      <Loader type="TailSpin" color="#5e454b" height={100} width={100} timeout={3000} />
    </div>
  );
};

export default Skelton;
