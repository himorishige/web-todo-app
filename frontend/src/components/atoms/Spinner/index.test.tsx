import renderer from 'react-test-renderer';
import Spinner from '.';

describe('Atoms/Spinner', () => {
  it('正しくレンダリングされている', () => {
    const element = renderer.create(<Spinner />).toJSON();
    expect(element).toMatchSnapshot();
  });
});
