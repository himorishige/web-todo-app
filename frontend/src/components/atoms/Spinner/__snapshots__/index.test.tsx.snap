// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Spinner 正しくレンダリングされている 1`] = `
<div>
  <div
    aria-busy="true"
    className=""
    style={Object {}}
  >
    <svg
      aria-label="audio-loading"
      height={100}
      viewBox="0 0 38 38"
      width={100}
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <linearGradient
          id="a"
          x1="8.042%"
          x2="65.682%"
          y1="0%"
          y2="23.865%"
        >
          <stop
            offset="0%"
            stopColor="#5e454b"
            stopOpacity="0"
          />
          <stop
            offset="63.146%"
            stopColor="#5e454b"
            stopOpacity=".631"
          />
          <stop
            offset="100%"
            stopColor="#5e454b"
          />
        </linearGradient>
      </defs>
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          transform="translate(1 1)"
        >
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            id="Oval-2"
            stroke="#5e454b"
            strokeWidth="2"
          >
            <animateTransform
              attributeName="transform"
              dur="0.9s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
          <circle
            cx="36"
            cy="18"
            fill="#fff"
            r={1}
          >
            <animateTransform
              attributeName="transform"
              dur="0.9s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </circle>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
